akka-projection-slick/src/test/scala/akka/projection/slick/SlickProjectionSpec.scala

```diff
@@ -357,15 +420,18 @@ class SlickProjectionSpec extends SlickSpec(SlickProjectionSpec.config) with Any
         offsetOpt shouldBe empty
       }
 
+      val eventHandler = new SlickEventHandler[Envelope] {
+        override def handleEvent(envelope: Envelope): slick.dbio.DBIO[Done] =
+          repository.concatToText(envelope.id, envelope.message)
+      }
       val slickProjection =
         SlickProjection.atLeastOnce(
           projectionId,
           sourceProvider = sourceProvider(entityId),
           databaseConfig = dbConfig,
           saveOffsetAfterEnvelopes = 1,
-          saveOffsetAfterDuration = Duration.Zero) { envelope =>
-          repository.concatToText(envelope.id, envelope.message)
-        }
+          saveOffsetAfterDuration = Duration.Zero,
+          eventHandler)
```

Comment:
I find that the event handler reads better then the lambda. The amount of arguments here is above my personal taste. Having a trailing lambda doesn't help.

Reply:
I think we should revisit a settings object using the builder pattern.  Most of these params could be included, I think the only exception might be the `sourceProvider` and `eventHandler` itself.